### PR TITLE
Updates libraries, tools and Gradle version

### PR DESCRIPTION
**Pull request by rafaeltoledo**
After closing #265, I played a little bit with Travis and now the build is passing with latest tools. Just made a small adjustement on lint to allow it to pass.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,10 +5,12 @@ jdk:
 
 android:
   components:
-    - platform-tools
     - tools
-    - android-23
-    - build-tools-23.0.3
+    - tools
+    - platform-tools
+    - extra
```
Comment (JakeWharton) [inline .travis.yml]: What is this for? It shouldn't be needed.

File: build.gradle [hunk 2]
```diff
@@ -7,17 +7,11 @@ buildscript {
...
-    classpath 'com.google.guava:guava:17.0'
-    classpath 'me.tatarka:gradle-retrolambda:3.2.5'
+    classpath 'com.android.tools.build:gradle:2.2.3'
+    classpath 'com.stanfy.spoon:spoon-gradle-plugin:1.2.2'
+    classpath 'me.tatarka:gradle-retrolambda:3.4.0'
     classpath 'me.tatarka.retrolambda.projectlombok:lombok.ast:0.2.3.a2'
   }
-
-  // Exclude the lombok version that the android plugin depends on.
   configurations.classpath.exclude group: 'com.android.tools.external.lombok'
```
Comment (JakeWharton) [inline build.gradle]: This can be removed.

File: build.gradle [hunk 3]
```diff
@@ -7,17 +7,11 @@ buildscript {
...
-    classpath('com.stanfy.spoon:spoon-gradle-plugin:1.0.3') {
-      // Workaround for https://github.com/stanfy/spoon-gradle-plugin/issues/33
-      exclude module: 'guava'
-    }
-    classpath 'com.google.guava:guava:17.0'
-    classpath 'me.tatarka:gradle-retrolambda:3.2.5'
+    classpath 'com.android.tools.build:gradle:2.2.3'
+    classpath 'com.stanfy.spoon:spoon-gradle-plugin:1.2.2'
+    classpath 'me.tatarka:gradle-retrolambda:3.4.0'
     classpath 'me.tatarka.retrolambda.projectlombok:lombok.ast:0.2.3.a2'
```
Comment (JakeWharton) [inline build.gradle]: This can be removed.

File: build.gradle [hunk 4]
```diff
@@ -112,7 +106,6 @@ android {
   lintOptions {
     textReport true
     textOutput 'stdout'
-    fatal 'UnusedResources'
```
Comment (JakeWharton) [inline build.gradle]: Revert

File: build.gradle (hunk 3 above)
Comment (rafaeltoledo) [inline build.gradle]: Removed as requested, but lint isn't handling lambdas in a proper way, even with latest plugin (the build it's still printing a stack trace)

**Comment by JakeWharton** [comment]
That's because libraries like Butter Knife and Timber need updated to use
the new lint PSI model instead of Lombok.

**Comment by rafaeltoledo** [comment]
Any new adjustement to allow this to be merged?

**Pull request merged**